sklearn/pipeline.py

```diff
@@ -831,3 +842,200 @@ def make_union(*transformers, **kwargs):
+                for name, trans, column in self.transformer_list
+                if not skip_none or trans is not None)
+
+    def _update_transformer_list(self, transformers):
+        transformers = iter(transformers)
+        self.transformer_list[:] = [
+            (name, None if old is None else next(transformers), column)
+            for name, old, column in self.transformer_list
+        ]
+
+    def fit(self, X, y=None):
+        """Fit all transformers using X.
+
+        Parameters
+        ----------
+        X : array-like or DataFrame of shape [n_samples, n_features]
+            Input data, of which specified subsets are used to fit the
+            transformers.
+
+        y : array-like, shape (n_samples, ...), optional
+            Targets for supervised learning.
+
+        Returns
+        -------
+        self : ColumnTransformer
+            This estimator
+        """
+        try:
+            return super(ColumnTransformer, self).fit(X, y=y)
+        except ValueError as e:
+            if "Got X with X.ndim=1. Reshape your data" in str(e):
+                raise ValueError(_ERR_MSG_1DCOLUMN)
+            else:
+                raise e
+
+    def fit_transform(self, X, y=None, **fit_params):
+        """Fit all transformers, transform the data and concatenate results.
+
+        Parameters
+        ----------
+        X : array-like or DataFrame of shape [n_samples, n_features]
+            Input data, of which specified subsets are used to fit the
+            transformers.
+
+        y : array-like, shape (n_samples, ...), optional
+            Targets for supervised learning.
+
+        Returns
+        -------
+        X_t : array-like or sparse matrix, shape (n_samples, sum_n_components)
+            hstack of results of transformers. sum_n_components is the
+            sum of n_components (output dimension) over transformers.
+        """
+        try:
+            return super(ColumnTransformer, self).fit_transform(X, y=y,
+                                                                **fit_params)
+        except ValueError as e:
+            if "Got X with X.ndim=1. Reshape your data" in str(e):
+                raise ValueError(_ERR_MSG_1DCOLUMN)
+            else:
+                raise e
+
+    def transform(self, X):
+        """Transform X separately by each transformer, concatenate results.
+
+        Parameters
+        ----------
+        X : array-like or DataFrame of shape [n_samples, n_features]
+            Input data, of which specified subsets are used to fit the
+            transformers.
+
+        Returns
+        -------
+        X_t : array-like or sparse matrix, shape (n_samples, sum_n_components)
+            hstack of results of transformers. sum_n_components is the
+            sum of n_components (output dimension) over transformers.
+        """
+        try:
+            return super(ColumnTransformer, self).transform(X)
+        except ValueError as e:
```

Comment:
This (and its sister below) hides the source of the actual transformer that raises, as far as I can tell. It might be preferable to just allow the original exception, until we have a good way to know what column needs to be reshaped.

Also, the check for "Got X with X.ndim..." will break with #9068